File: kotlin-springboot-jpa/app/src/main/java/es/msanchez/frameworks/spring/boot/service/PersonService.java
package es.msanchez.frameworks.spring.boot.service;

import es.msanchez.frameworks.spring.boot.dao.PersonDao;
import es.msanchez.frameworks.spring.boot.entity.Person;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Slf4j
@Service
public class PersonService {

    private final PersonDao personDao;

    public PersonService(final PersonDao personDao) {
        this.personDao = personDao;
    }

    public boolean isValid(final String name) {
        final Optional<Person> exists = this.personDao.findOneByName(name);
        return !exists.isPresent();
    }

    public void save(final Person person) {
        this.personDao.save(person);
    }

}

File: kotlin-springboot-jpa/app/src/main/kotlin/es/msanchez/frameworks/spring/boot/service/PersonService.kt
package es.msanchez.frameworks.spring.boot.service

import es.msanchez.frameworks.spring.boot.dao.PersonDao
import es.msanchez.frameworks.spring.boot.entity.Person
import org.springframework.stereotype.Service

@Service
class PersonService(private val personDao: PersonDao) {

    fun isValid(name: String): Boolean {
        val person = this.personDao.findOneByName(name)
        return !person.isPresent
    }

    fun save(person: Person) {
        this.personDao.save(person)
    }

}